k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp

```diff
@@ -209,14 +209,16 @@ StatusCode MarlinProcessorWrapper::execute() {
 
   if (sc.isFailure()) {
     // Register empty event
-    the_event = new lcio::LCEventImpl();
     debug() << "Registering empty Event for EDM4hep to LCIO conversion event in TES" << endmsg;
-    auto       pO     = std::make_unique<LCEventWrapper>(the_event, true);
+    auto       pO     = std::make_unique<LCEventWrapper>(std::make_unique<lcio::LCEventImpl>());
     StatusCode reg_sc = evtSvc()->registerObject("/Event/LCEvent", pO.release());
     if (reg_sc.isFailure()) {
       error() << "Failed to register empty LCIO Event" << endmsg;
       return reg_sc;
     }
+    DataObject* eventObject = nullptr;
+    eventSvc()->retrieveObject("/Event/LCEvent", eventObject).ignore();
```

Comment:
I suppose ignoring the StatusCode here is something we can do, because just above we make sure that something is actually there?

Reply:
My experience with this is not large enough to know if that can go wrong :)

I think getting `theEvent` out of `pO` before the `registerObject` is more straightforward?
`theEvent = static_cast<IMPL::LCEventImpl*>(pO->getEvent());`

Reply:
Yeah, that is probably more straight forward, and one less thing to worry about. Will do that